rfc-capi-for-kvm-capk/README.md

```diff
@@ -0,0 +1,425 @@
+- As a GS PE, I want KVM operators to work in arbitrary Kubernetes clusters so we can stop maintaining our custom provisioning tooling for on-prem management clusters.
+- As a GS sales person, I want KVM operators to implement the CAPI provider contract so that we can be in the list of CAPI providers and find other potential customers.
+- As a GS on-premises (on-prem) customer, I want KVM operator to implement CAPI so that I have access to new provider-independent features developed by other GS teams.
+- As a GS oncall engineer, I want on-prem installations to be as similar as possible to the other providers so that I can share best practices, on call routines, and documentation with the other teams.
+- As a GS on-prem customer, I want GS to implement a migration strategy from current kvm clusters to CAPI KVM clusters so that I can access the new features without service disruption and workload migration.
+- As a GS on-prem customer, I want GS to implement a break out/migration strategy for on-prem installations so that I can use other CAPI on-prem providers in the future such as CAPV (Cluster API for VSphere (VMWare)) or metal3 (Cluster API for bare metal).
+- As a member of Team Rocket, I would like to have confidence in my changes to the platform through the use of reliable unit, integration, and e2e tests.
+
+
+### Rationale
+
+With the current move to full CAPI adoption in Team Firecracker and Team Celestial, Team Rocket would be the only user and therefore the only maintainer of the current components and code that would be legacy for the other teams.
+
+This would mean that we would need to put a lot of effort into understanding and owning this code fully in order to be able to fix bugs and maintain KVM installations
+in the future.
+
+Instead of being the sole maintainer of a huge amount of legacy code, Team Rocket wants to invest this time in implementing CAPI for KVM operator and maintain relative platform parity with the other GS provider teams.
+
+### Goals
+
+- Customers can upgrade existing clusters to CAPK without significant workload downtime.
+- Existing CAPI-compatible tools such as kubectl-gs, happa, GS api, and clusterctl work with CAPK clusters.
+- Team Rocket is responsible for a significantly reduced number of repos and lines of code.
+
+### Non-goals
+
+- Get CAPK accepted as a CNCF project.
+- Support mixed CAPK/CAPV clusters.
+- Support mixed management clusters (e.g. CAPK on AWS).
+- Support `MachineSet` or `MachinePool`.
+- Support CAPI <=v1alpha3.
+
+### Scope
+
+To minimize the scope, we will start with a minimal feature set and add only features that are used by the current customers. Features that are currently not used by any customer will be dropped.
+
+Current features of `kvm-operator` (with a ✅ if we decided to keep it in CAPI):
```

Comment:
This is a bit confusing - most of these which are related to the api-server / kubelet might still be covered by the general CAPI controllers IMO.

So would the feature just move?

Reply:
Yes, most of them will go in the `KubeadmConfig` I think.